addons/knowledge/static/src/js/knowledge_controller.js

```diff
@@ -221,30 +260,51 @@ const KnowledgeFormController = FormController.extend({
     },
 
     /**
-     * @param {integer} src
-     * @param {integer} dst
+     * @param {Object} data
+     * @param {integer} data.article_id
+     * @param {String} data.category
+     * @param {integer} [data.target_parent_id]
+     * @param {integer} [data.before_article_id]
+     * @param {Function} data.onSuccess
+     * @param {Function} data.onReject
      */
-    _move: async function (src, dst) {
+    _move: async function (data) {
+        if (data.category === 'shared') {
+            // TODO: What should happen when the user drags an article in 'shared' ?
+            data.onReject();
```

Comment:
Well, we cannot move an article at shared root, unless the article is already in shared category.
So we should not even add this item in the dropdown at all when conditions are not met, instead of allowing to select it, then reject it.